### PR TITLE
Gives the top contributor a coder skirt

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Overrides jobs clothes for the #1 contributor and gives them the coder skirt.
image
Its animated and everything.

## Why It's Good For The Game
to show our appreciation for whoever did the most PRs.

## Changelog
:cl:
add: Coderskirt
refactor: Refactors get contributors code to only get the list once and store it as a static variable.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
So who is the top contributor?

File: code/modules/clothing/under/skirt_dress.dm [hunk 1]
```diff
@@ -99,3 +99,19 @@
...
+/obj/item/clothing/under/dress/skirt/coder
+	name = "coder skirt"
+	desc = "The best of the best. The ultimate femboy. Many have stood before them, however they were powerful enough to not only wear the socks, but to embrace them."
+	icon_state = "coderskirt"
+	item_state = "plaid_green"
+	item_color = "coderskirt"
+
+/obj/item/clothing/under/dress/skirt/coder/Initialize()
+	. = ..()
+	desc = "It has a tag on it reading \'Pull request approved by [get_top_contrib()].\''"
```
Comment (PowerfulBacon) [inline code/modules/clothing/under/skirt_dress.dm]:
```suggestion
	desc = "It has a tag on it reading \'Pull request approved by [sanitize(get_top_contrib())].\''"
```

**Comment by PowerfulBacon** [comment]
> 
> 
> So who is the top contributor?

I have literally no idea, check the credits and its the name at the top.

**Comment by qwertyquerty** [comment]
Damn time to make more PRs

**Review by qwertyquerty — APPROVED** [review]

**Comment by MCterra10** [comment]
who thought this was a good idea

**Comment by ike709** [comment]
God damn it qwerty.

File: code/_onclick/hud/credits.dm [hunk 2]
```diff
@@ -83,13 +83,18 @@ GLOBAL_LIST(end_titles)
...
-/proc/get_contribs()
-	var/list/contribs = list()
+/proc/get_top_contrib()
+	var/list/contribs = get_contribs()
+	return contribs.len ? contribs[1] : "Beebot"
 
-	if(fexists("[global.config.directory]/contributors.txt"))
-		contribs += world.file2list("[global.config.directory]/contributors.txt")
+/proc/get_contribs()
+	var/static/list/top_contributors
```
Comment (ike709) [inline code/_onclick/hud/credits.dm:91]: why is this a static list

File: code/modules/clothing/under/skirt_dress.dm [hunk 3]
```diff
@@ -99,3 +99,19 @@
...
+/obj/item/clothing/under/dress/skirt/coder
+	name = "coder skirt"
+	desc = "The best of the best. The ultimate femboy. Many have stood before them, however they were powerful enough to not only wear the socks, but to embrace them."
+	icon_state = "coderskirt"
+	item_state = "plaid_green"
+	item_color = "coderskirt"
+
+/obj/item/clothing/under/dress/skirt/coder/Initialize()
+	. = ..()
+	desc = "It has a tag on it reading \'Pull request approved by [sanitize(get_top_contrib())].\''"
```
Comment (ike709) [inline code/modules/clothing/under/skirt_dress.dm:112]: Bruh note the extra `'`

File: code/modules/jobs/job_types/_job.dm [hunk 4]
```diff
@@ -184,6 +184,15 @@
 
 	H.dna.species.after_equip_job(src, H, visualsOnly)
 
+	if(preference_source?.ckey == get_top_contrib())
```
Comment (ike709) [inline code/modules/jobs/job_types/_job.dm:187]: Bruh

File: code/modules/jobs/job_types/_job.dm [hunk 5]
```diff
@@ -184,6 +184,15 @@
...
+		var/obj/item/clothing/under/dress/skirt/coder/coderskirt = new()
```
Comment (ike709) [inline code/modules/jobs/job_types/_job.dm:188]: Don't spawn it in nullspace. Spawn it on the person's loc.

File: code/modules/jobs/job_types/_job.dm [hunk 6]
```diff
@@ -184,6 +184,15 @@
...
+		if(H.wear_suit)
+			H.dropItemToGround(H.wear_suit)
+		if(!H.equip_to_slot_if_possible(coderskirt, SLOT_WEAR_SUIT))
+			to_chat(H, "<span class='warning'>You failed to don your favorite skirt this morning.</span>")
+		else
+			to_chat(H, "<span class='notice'>Roses are red, violets are blue; You are the top contributer, fem energy flows through you.</span>")
```
Comment (ike709) [inline code/modules/jobs/job_types/_job.dm:194]: Typo

**Review by ike709 — COMMENTED** [review]
Shouldn't even be a thing on Sage.

**Comment by Ratvarr** [comment]
it should

**Comment by zeskorion** [comment]
what if you're a plamsmaman

**Pull request merged**